server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputerTests.java

```diff
@@ -616,42 +621,57 @@ public void testComputeConsideringShardSizes() {
 
         var discoveryNodesBuilder = DiscoveryNodes.builder()
             .add(createDiscoveryNode("node-0", DiscoveryNodeRole.roles()))
-            .add(createDiscoveryNode("node-1", DiscoveryNodeRole.roles()));
+            .add(createDiscoveryNode("node-1", DiscoveryNodeRole.roles()))
+            .add(createDiscoveryNode("node-2", DiscoveryNodeRole.roles()));
 
         var metadataBuilder = Metadata.builder();
         var routingTableBuilder = RoutingTable.builder();
 
-        for (int i = 0; i < 10; i++) {
+        var commonIndexSettings = Settings.builder()
+            .put("index.number_of_shards", 1)
+            .put("index.number_of_replicas", 0)
```

Comment:
I find it slightly confusing that we use 0 replicas for an index with replicas. I assume it works because we do not depend on it really, but I think I'd prefer to have this be a set explicitly to 1 for "index-0".